include/mtx/responses/public_rooms.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#if __has_include(<nlohmann/json_fwd.hpp>)
+#include <nlohmann/json_fwd.hpp>
+#else
+#include <nlohmann/json.hpp>
+#endif
+
+namespace mtx {
+namespace responses {
+
+//! Whether or not the room will be visible by non members.
+enum class Visibility
+{
+        //! A private visibility will hide the room from the published room list.
+        Private,
+        //! Indicates that the room will be shown in the published room list.
+        Public,
+};
+
+inline std::string
+visibilityToString(Visibility visibility)
+{
+        if (visibility == Visibility::Private) {
+                return "private";
+        }
+
+        return "public";
+}
+
+inline Visibility
+stringToVisibility(const std::string &s)
+{
+        if (s == "private") {
+                return Visibility::Private;
+        }
+        return Visibility::Public;
+}
+
+//! Response from the `GET /_matrix/client/r0/directory/list/room/{roomId}`endpoint.
+struct RoomVisibility
+{
+        //! The visibility of the room in the directory. One of: ["private", "public"]
+        Visibility visibility;
+};
```

Comment:
I don't really like, that this is repeated form requests.hpp. Can you remove it there and put this in mtx::common/common.hpp or so?

Reply:
done!

Reply:
Perfect, thanks!